packages/utils/src/string/index.ts

```
@@ -0,0 +1,834 @@
+
+function cached(fn) {
+  let cache = Object.create(null)
+
+  return function cachedFn(str) {
+    const hit = cache[str]
+    return hit || (cache[str] = fn(str))
+  }
+}
+
+const camelizeRE = /-(\w)/g
+export const camelize = cached((str) => str.replace(camelizeRE, (_, c) => (c ? c.toUpperCase() : '')))
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)